akismet-api.d.ts

```diff
@@ -0,0 +1,63 @@
+declare module "akismet-api" {
+  // Type definitions for akismet-api 5.2.1
+  // Definitions by: Jason Florentino https://github.com/jasonflorentino
+
+  // Main Akismet Client Class
+  export class AkismetClient {
+    constructor(options: AkismetClientOptions);
+
+    verifyKey(callback?: () => void): void;
+
+    checkSpam(comment: Comment, callback?: () => void): boolean;
+    submitSpam(comment: Comment, callback?: () => void): void;
+    submitHam(comment: Comment, callback?: () => void): void;
+  }
+
+  // Legacy props
+  export type Client = AkismetClient;
+  export type client = (options: AkismetClientOptions) => AkismetClient;
+
+  // Types
+  interface AkismetClientOptions {
+    blog: string;
+    key: string;
+    host?: string;
+    protocol?: string;
+    version?: string;
+    userAgent?: string;
+  
+    blog_lang?: string;
+    lang?: string; // Alias for prev prop 
+    
+    blog_charset?: string;
+    charset?: string; // Alias for prev prop 
+  }
+
+  // Require only one of these two types
+  type Comment = (CommentWithIP) | (CommentWithUserIP);
+
+  // BaseComment with either `ip` or `user_up`
+  type CommentWithIP = { ip: string; } & BaseComment;
+  type CommentWithUserIP = { user_ip: string; } & BaseComment;
+
+  // BaseComment is composed of Allowed Keys, all optional
+  type BaseComment = 
+    Partial<Record<CommentKeys, string>> 
+    & Partial<Record<TestKeys, boolean | string>>;
+
+  // Allowed Keys
+  type TestKeys = 'is_test' | 'isTest';
+  
+  type CommentKeys = 'user_ip' | 'ip'
+    | 'user_agent' | 'useragent'
```

Comment:
I thought that user agent was required but this seems to no longer be the case. Our documentation is therefore out of date, so I've created https://github.com/chrisfosterelli/akismet-api/issues/198. I think this implementation is good.

Reply:
Sweet! Thanks!